cli/src/git_util.rs

```diff
@@ -456,30 +466,36 @@ export or their "parent" bookmarks."#,
     Ok(())
 }
 
+pub struct FetchArgs<'a> {
+    pub repo: &'a git2::Repository,
```

Comment:
I wouldn't include `git2::Repository` in args struct because it's not a user parameter and has different lifetime.

BTW, what I meant was to extract common `clap::Args` struct from `GitFetchArgs`. If `jj git sync` shares the same help messages, we can "flatten" the common Args struct into `GitFetch`/`SyncArgs`. If not, we'll need a separate parameter struct as you did.

Reply:
> I wouldn't include `git2::Repository` in args struct because it's not a user parameter and has different lifetime.

Removed.

> 
> BTW, what I meant was to extract common `clap::Args` struct from `GitFetchArgs`. If `jj git sync` shares the same help messages, we can "flatten" the common Args struct into `GitFetch`/`SyncArgs`. If not, we'll need a separate parameter struct as you did.

I've gone to flesh out a first draft of the sync command to be sure we need to share the help messages and yeah, it makes sense to share them. So I've extracted these out and flattened them back into `GitFetchArgs`.

Is this what you meant?